src/components/fx/hover.js

```diff
@@ -1095,7 +1096,13 @@ function cleanPoint(d, hovermode) {
         d.yVal = d.ya.c2d(d.yLabelVal);
     }
 
-    if(d.zLabelVal !== undefined) d.zLabel = String(d.zLabelVal);
+    if(d.zLabelVal !== undefined) {
+        if(d.zformat !== undefined) {
+            d.zLabel = d3.format(d.zformat)(d.zLabelVal).replace(/-/g, constants.MINUS_SIGN);
+        } else {
+            d.zLabel = String(d.zLabelVal);
```

Comment:
I would prefer adding the new formatting logic in the trace `hoverPoints` modules - e.g. this [one](https://github.com/plotly/plotly.js/blob/f1faa2871dbf182847579166af47d1cc2a983707/src/traces/heatmap/hover.js#L112).